typing/typedecl.ml

```diff
@@ -979,7 +979,8 @@ let transl_declaration env sdecl (id, uid) =
           Ttype_record_unboxed_product lbls,
           Type_record_unboxed_product(lbls', Record_unboxed_product), jkind
       | Ptype_open ->
-        Ttype_open, Type_open, Jkind.Builtin.value ~why:Extensible_variant
+        Ttype_open, Type_open,
+        (Jkind.Builtin.value ~why:Extensible_variant |> Jkind.mark_best)
```

Comment:
Why this change? I think it's probably sound, because extensible types never get better kinds, but I think it's a nice property to have all jkinds used in the midst of type-checking a declaration group to have non-best kinds. At the least, I think there should be a nice comment here explaining why we do this. (And, come to think of it, additions to the big comment about jkind defaults describing how quality flows through the algorithm.)

Reply:
It turns out we don't need this, so I just got rid of it.